### PR TITLE
[KNUDEV-15] Added intergrations tests for account registration

### PR DESCRIPTION
**Pull request by cumbainer**

### CODE REVIEW

File: knudev-app/src/main/resources/application.yml [hunk 1]
```diff
@@ -15,6 +15,8 @@ spring:
     change-log: classpath:db/master.yaml
   jackson:
     time-zone: Europe/Kiev
+  jpa:
+    show-sql: true
```
Comment (DenysLnk) [inline knudev-app/src/main/resources/application.yml:19]: Delete it

Comment (DenysLnk) [reply knudev-app/src/main/resources/application.yml:19]: Or comment

File: knudev-common/src/main/java/ua/knu/knudev/knudevcommon/utils/FullName.java [hunk 2]
```diff
@@ -1,11 +1,21 @@
 package ua.knu.knudev.knudevcommon.utils;
 
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Pattern;
 import lombok.Builder;
 
 @Builder
 public record FullName(
+        @NotBlank(message = "First name cannot be null or blank.")
+        @Pattern(regexp = "^[Ѐ-ӿ'-]+$", message = "First name must contain only Ukrainian characters and valid symbols (- or ')")
```
Comment (DenysLnk) [inline knudev-common/src/main/java/ua/knu/knudev/knudevcommon/utils/FullName.java]: We told about name in English

File: knudev-security-api/src/main/java/ua/knu/knudev/knudevsecurityapi/request/AccountCreationRequest.java [hunk 3]
```diff
@@ -23,11 +25,20 @@ public record AccountCreationRequest(
         @Size(
                 min = 5,
                 max = 64,
-                message = "Password must be between 8 and 64 characters"
+                message = "Password must be between 5 and 64 characters"
```
Comment (DenysLnk) [inline knudev-security-api/src/main/java/ua/knu/knudev/knudevsecurityapi/request/AccountCreationRequest.java]:
```suggestion
                message = "Password must be between 8 and 64 characters"
```

File: knudev-security-api/src/main/java/ua/knu/knudev/knudevsecurityapi/request/AccountCreationRequest.java [hunk 4]
```diff
@@ -23,11 +25,20 @@ public record AccountCreationRequest(
...
         )
+        @Pattern(
+                regexp = "^(?=.*[a-zA-Z])(?=.*\\d).*$",
+                message = "Password must contain at least one letter and one digit"
+        )
+
```
Comment (DenysLnk) [inline knudev-security-api/src/main/java/ua/knu/knudev/knudevsecurityapi/request/AccountCreationRequest.java]:
```suggestion
```

File: knudev-security-api/src/main/java/ua/knu/knudev/knudevsecurityapi/request/AccountCreationRequest.java [hunk 5]
```diff
@@ -23,11 +25,20 @@ public record AccountCreationRequest(
...
         String password,
+
```
Comment (DenysLnk) [inline knudev-security-api/src/main/java/ua/knu/knudev/knudevsecurityapi/request/AccountCreationRequest.java:35]:
```suggestion
```

File: team-manager-intergration-tests/src/test/java/ua/knu/knudev/intergrationtests/utils/constants/AccountTestsConstants.java [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+import ua.knu.knudev.knudevsecurityapi.constant.AccountRole;
+
+public class AccountTestsConstants {
+
+    public static final String TEST_EMAIL = "testKnuDevEmail@knu.ua";
+    public static final String TEST_PASSWORD = "qwerty12345";
+    public static final String TEST_FILE_NAME = "testFilename";
+    public static final AccountRole TEST_ROLE = AccountRole.INTERN;
+
+    public static final String PROFILE_FIRST_NAME = "Джон";
```
Comment (DenysLnk) [inline team-manager-intergration-tests/src/test/java/ua/knu/knudev/intergrationtests/utils/constants/AccountTestsConstants.java]: Lets use English nameing

File: team-manager/src/main/java/ua/knu/knudev/teammanager/service/AccountProfileService.java [hunk 7]
```diff
@@ -1,55 +1,86 @@
...
 import ua.knu.knudev.teammanagerapi.exception.AccountException;
 import ua.knu.knudev.teammanagerapi.response.AccountRegistrationResponse;
 
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+
 @Service
 @RequiredArgsConstructor
+@Validated
```
Comment (DenysLnk) [inline team-manager/src/main/java/ua/knu/knudev/teammanager/service/AccountProfileService.java:35]: Will discuss it, wetold that it doesn't work in service module

File: team-manager/src/main/java/ua/knu/knudev/teammanager/service/AccountProfileService.java [hunk 8]
```diff
@@ -1,55 +1,86 @@
...
+        boolean accountProfileExists = accountProfileRepository.existsByEmail(email);
+        if (accountProfileExists) {
             throw new AccountException(
                     String.format("Account with email %s already exists", email)
             );
         }
+
+        boolean accountAuthExists = accountAuthServiceApi.existsByEmail(email);
+        if (accountAuthExists) {
+            log.error("AccountProfile does not exists, but AccountAuth does exist by email {}", email);
```
Comment (DenysLnk) [inline team-manager/src/main/java/ua/knu/knudev/teammanager/service/AccountProfileService.java:81]: You will throw exception with message already, so you don't need message

File: team-manager/src/main/java/ua/knu/knudev/teammanager/service/AccountProfileService.java [hunk 9]
```diff
@@ -74,18 +105,39 @@ private AccountProfile buildAccountProfile(AccountCreationRequest request,
...
+                .registrationDate(LocalDateTime.now())
+                .expertise(request.expertise())
+                .roles(new HashSet<>(authAccount.roles()))
                 .build();
     }
 
     private AccountRegistrationResponse buildRegistrationResponse(AccountProfile savedAccount,
-                                                                  String requestEmail) {
-        AccountProfileDto account = accountProfileMapper.toDto(savedAccount);
+                                                                  String requestEmail,
```
Comment (DenysLnk) [inline team-manager/src/main/java/ua/knu/knudev/teammanager/service/AccountProfileService.java:118]: Don't need write 1 property on 1 line, write as I told you, it'll be better

File: team-manager/src/test/java/ua/knu/knudev/teammanager/service/AccountProfileServiceTest.java [hunk 10]
```diff
@@ -96,15 +99,16 @@ void setUp() {
 
     @Test
     @DisplayName("Should register successfully when all inputs are valid")
+    @SneakyThrows
```
Comment (DenysLnk) [inline team-manager/src/test/java/ua/knu/knudev/teammanager/service/AccountProfileServiceTest.java]: You don't need it here

**Review by DenysLnk — CHANGES_REQUESTED** [review]

File: knudev-common/src/main/java/ua/knu/knudev/knudevcommon/utils/FullName.java (hunk 2 above)
Comment (cumbainer) [reply knudev-common/src/main/java/ua/knu/knudev/knudevcommon/utils/FullName.java]: remade to eng

File: team-manager-intergration-tests/src/test/java/ua/knu/knudev/intergrationtests/utils/constants/AccountTestsConstants.java (hunk 6 above)
Comment (cumbainer) [reply team-manager-intergration-tests/src/test/java/ua/knu/knudev/intergrationtests/utils/constants/AccountTestsConstants.java]: done

File: team-manager/src/main/java/ua/knu/knudev/teammanager/service/AccountProfileService.java (hunk 7 above)
Comment (cumbainer) [reply team-manager/src/main/java/ua/knu/knudev/teammanager/service/AccountProfileService.java:35]: it does

File: team-manager/src/main/java/ua/knu/knudev/teammanager/service/AccountProfileService.java (hunk 8 above)
Comment (cumbainer) [reply team-manager/src/main/java/ua/knu/knudev/teammanager/service/AccountProfileService.java:81]: This is very rare case. It ideally would never happen, but if it does - we will see the logs

File: team-manager/src/main/java/ua/knu/knudev/teammanager/service/AccountProfileService.java (hunk 9 above)
Comment (cumbainer) [reply team-manager/src/main/java/ua/knu/knudev/teammanager/service/AccountProfileService.java:118]: done

File: team-manager/src/test/java/ua/knu/knudev/teammanager/service/AccountProfileServiceTest.java (hunk 10 above)
Comment (cumbainer) [reply team-manager/src/test/java/ua/knu/knudev/teammanager/service/AccountProfileServiceTest.java]: Rewrote to try catch
I have getBytes method, it throws IOException

**Pull request merged**